src/pages/index.js

```diff
@@ -74,54 +104,78 @@ const cardTitleInput = addCardFormElement.querySelector(
 );
 const cardURLInput = addCardFormElement.querySelector(".modal__input_type_url");
 
-function closeWithEsc(event) {
-  if (event.key === "Escape") {
-    const modal = document.querySelector(".modal_opened");
-    closeModal(modal);
-  }
-}
-
-function closeModalOnRemoteClick(event) {
-  if (event.target === event.currentTarget) {
-    closeModal(event.currentTarget);
-  }
-}
-
-function openModal(modal) {
-  // add class to modal
-  document.addEventListener("keydown", closeWithEsc);
-  modal.addEventListener("mousedown", closeModalOnRemoteClick);
-  modal.classList.add("modal_opened");
-}
-
-function closeModal(modal) {
-  //remove class from modal
-  document.removeEventListener("keydown", closeWithEsc);
-  modal.removeEventListener("mousedown", closeModalOnRemoteClick);
-  modal.classList.remove("modal_opened");
-}
+const pictureURLInput = pictureFormElement.querySelector(
+  ".modal__input_type_url"
+);
 
 function renderCard(cardData) {
   const card = createCard(cardData);
   section.addItem(card);
-  //wrapper.prepend();
-  //Section.renderItems(item);
 }
 
 function handleImagePreview(cardData) {
-  popupImage.open(cardData); 
+  popupImage.open(cardData);
+}
+
+function handleSubmit(request, popupInstance, form, loadingText = "Saving...") {
+  popupInstance.renderLoading(true, loadingText);
+  // setTimeout(() => {
+  request()
+    .then(() => {
+      popupInstance.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+      if (form) {
+        form.reset();
```

Comment:
```js

      if (form) {
        form.reset();
      }
```

It should be in `then`

You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.



```js

  request()
    .then(() => {
      popupInstance.close();
      if (form) {
        form.reset();  // here
      }
    })
    .catch(console.error)
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}

```